input/cursor: correctly transfer focus when using tablet pen

Xyene opened the pull request:
Tentatively fixes #4819.

This commit ensures that `seatop_motion` is called to transfer focus when a window is selected via a pen. Previously, it would race with `node_at_coords`, and only properly transfer focus if its returned `surface` was `NULL`.

I've tested this locally, and it fixes the issues I was experiencing.

Calling `seatop_motion` directly here works as well, but my impression is that the confine handling inside `cursor_motion` should be done for tablet devices too. Applications see the following sequence of events:

```
[912231.664] wl_pointer@3.motion(7987491, 281.539062, 991.007812)
[912231.673] zwp_tablet_tool_v2@4278190083.motion(281.539062, 991.007812)
[912231.676] zwp_tablet_tool_v2@4278190083.frame(7987491)
```

In a GNOME session on the same application, the `wl_pointer` event is not present. I don't know enough to say which is the correct behaviour, or if it even matters. Sway infrequently omits the `wl_pointer` event, too.

emersion commented:
>Calling seatop_motion directly here works as well, but my impression is that the confine handling inside cursor_motion should be done for tablet devices too.

Yes. The `wl_pointer.motion` event is incorrect.

Xyene commented:
Thanks! So, to confirm:

* We don't want `wlr_relative_pointer_manager_v1_send_relative_motion` to be called for tablet events.
* We _do_ want `wlr_region_confine` to be called.
* We don't want `wlr_cursor_move` to be called. (Is this what triggers `wl_pointer.motion`?)

In that case, a follow-up question is: is https://github.com/swaywm/sway/blob/fcd524bb0d6f3c6cd4f2ddc180520bc56a89aa4a/sway/input/cursor.c#L481-L484 correct? The branch is taken even if the surface supports tablet v2, if the `surface` returned by `node_at_coords` is `NULL`, and ends up sending `wl_pointer.motion`. That branch is how tablet focus seems to work right now (even if erratically).

emersion commented:
>and ends up sending wl_pointer.motion

Just like touch, we should use `seat_set_focus` to transfer focus.

Xyene commented:
I've updated this PR to call `seat_set_focus`. I also added a check for `focus_follows_mouse`, which while not semantically correct (perhaps a new option should be better), matches i3 behavior.
